powersimdata/input/check.py

```diff
@@ -210,3 +222,362 @@ def _check_connected_components(grid, error_messages):
             f"but is specified as having {num_interconnects} interconnects: "
             f"{grid.interconnect}."
         )
+
+
+def _check_data_frame(df, label):
+    """Ensure that input is a pandas data frame.
+    :param pandas.DataFrame df: a data frame.
+    :param str label: name of data frame (used for error messages).
+    :raises TypeError: if df is not a data frame or label is not a str.
+    :raises ValueError: if data frame is empty.
+    """
+    if not isinstance(label, str):
+        raise TypeError("label must be a str")
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError(label + " must be a pandas.DataFrame object")
+    if not df.shape[0] > 0:
+        raise ValueError(label + " must have at least one row")
+    if not df.shape[1] > 0:
+        raise ValueError(label + " must have at least one column")
+
+
+def _check_time_series(ts, label):
+    """Check that a time series is specified properly.
+    :param pandas.DataFrame/pandas.Series ts: time series to check.
+    :param str label: name of time series (used for error messages).
+    :raises TypeError: if ts is not a data frame/time series or label is not a str.
+    :raises ValueError: if indices are not timestamps.
+    """
+    if not isinstance(label, str):
+        raise TypeError("label must be a str")
+    if not isinstance(ts, (pd.DataFrame, pd.Series)):
+        raise TypeError(label + " must be a pandas.DataFrame or pandas.Series object")
+    if not ts.shape[0] > 0:
+        raise ValueError(label + " must have at least one row")
+    if not isinstance(ts.index, pd.DatetimeIndex):
+        raise ValueError(label + " must be a time series")
+
+
+def _check_grid_type(grid):
+    """Ensure that ``grid`` is a Grid object.
+    :param powersimdata.input.grid.Grid grid: a Grid instance.
+    :raises TypeError: if input is not a Grid instance.
+    """
+    if not isinstance(grid, Grid):
+        raise TypeError("grid must be a powersimdata.input.grid.Grid object")
```

Comment:
Could use `f"grid must be a {Grid} object"` to avoid hard coding the type. 
It will print out `<class 'powersimdata.input.grid.Grid'>`, so still readable.

Reply:
I guess a similar comment would apply to the various places we raise `TypeError`

Reply:
That's clever, I like it.

Reply:
Done.